component/http/v2/check_test.go

```diff
@@ -0,0 +1,62 @@
+package v2
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_aliveCheckRoute(t *testing.T) {
+	tests := map[string]struct {
+		acf  LivenessCheckFunc
+		want int
+	}{
+		"alive":        {func() AliveStatus { return Alive }, http.StatusOK},
+		"unresponsive": {func() AliveStatus { return Unhealthy }, http.StatusServiceUnavailable},
+		"default":      {func() AliveStatus { return 10 }, http.StatusOK},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			route := LivenessCheckRoute(tt.acf)
```

Comment:
These tests could all be parallelized but I seem to remember that we do not want to enable that now.

Reply:
done